test/integration/controller_test.go

```diff
@@ -696,6 +737,16 @@ type controllerTest struct {
 	// reconciliation is expected to fail.
 	skipVerifyingBindingSuccess bool
 
+	skipBrokerDelete bool
```

Comment:
I would rather see the resources brought into a state where they can be deleted successfully prior to tearing down the tests instead of skipping the delete success check. That way we are adding some additional checks that verify that we are able to recover from failed states successfully.

For example, in the "provisioning with last operation response state in progress" test case in `TestPollServiceInstanceLastOperation`, I would prefer to see the following test behavior.

1. The controller polls for the last operation a few times with a response state of InProgress.
1. After a few polls of the last operation, the broker reaction is changed to provide a response state of Succeeded.

By the way, there is already a test to do exactly that in `TestAsyncProvisionWithMultiplePolls`.

Reply:
That's a clever solution that I hadn't seen. And I agree it feels icky to shutdown the test without everything being properly deleted... Maybe there needs to be a deferred delete? I'm using `skipBindingDelete` in the "deprovision instance with binding not deleted" test.

Reply:
Let's take that same test case and expand it so that there was something that was blocking a successful delete of the binding. For example, let's say that the broker was setup to respond with failure to the request to the delete the binding. Once the test did the check to verify that the instance deprovision was blocked, the test can change the broker response so that subsequent delete-binding requests succeeded, thereby freeing the binding and instance to be deleted successfully.